src/Support/Generators/RelationPropertyGenerator.php

```diff
@@ -33,8 +33,16 @@ public function getPropertyDefinition(Model $model): array
         $withProperty->setAccessible(true);
         $withFields = $withProperty->getValue($model);
 
+        $permittedClasses = [get_class($model)];
+        foreach(class_parents($model) as $parent) {
+            if ($parent == Model::class) {
+                break;
+            }
+            $permittedClasses[] = $parent;
+        }
+
```

Comment:
Can you add some tests for this case?

Reply:
I have given it a try, and almost got it, but realised that the code in `src/Support/Generators/RelationPropertyGenerator.php` line `61` (where it checks that the related class starts with `App`) blocks a test class. Would need to unblock that for a test to run, which may have other implications on the generation.